theano/sandbox/gpuarray/dnn.py

```diff
@@ -1783,6 +1395,22 @@ def local_softmax_dnn(node):
         return [out]
 
 
+@register_opt('cudnn')
+@local_optimizer([GpuElemwise])
+def local_log_softmax_dnn(node):
+    if not dnn_available() or version() < 3000:
+        # No log-softmax before cudnn v3
+        return
+    if (isinstance(node.op, GpuElemwise) and
+            isinstance(node.op.scalar_op, Log) and
+            node.inputs[0].owner and
+            isinstance(node.inputs[0].owner.op, GpuDnnSoftmax) and
+            len(node.inputs[0].clients) == 1):
+        softmax_node = node.inputs[0].owner
+        new_softmax = GpuDnnSoftmax(None, 'log', softmax_node.op.mode)
+        return [new_softmax(softmax_node.inputs[0])]
+
+
```

Comment:
I was certain I added tests for that, but apparently I didn't.

I'll copy the tests from the old backend.